app/src/main/java/org/mozilla/fenix/share/SaveToPDFMiddleware.kt

```diff
@@ -36,9 +34,10 @@ class SaveToPDFMiddleware(
         if (action is EngineAction.SaveToPdfExceptionAction) {
             // See https://github.com/mozilla-mobile/fenix/issues/27649 for more details,
             // why a Toast is used here.
-            GlobalScope.launch(Dispatchers.Main) {
+            ThreadUtils.runOnUiThread {
```

Comment:
@DreVla to make test happy, I removed launch from your changes and kept the `ThreadUtils.runOnUiThread`.